### ISSUE TITLE
[Question] How can I create Immediates?

### ISSUE THREAD

**Issue by TornaxO7** [icedland/iced#337]
Hello! I'm wondering how I can execute a `SUB` with an immediate and a register. I couldn't find anything in the docs, could you please help me here?

## Minimal example
`Cargo.toml`:
```toml
[package]
name = "yeet"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]

[dependencies.iced-x86]
version = "1.17.0"
default-features = false
features = ["encoder", "code_asm", "std"]
```

`src/main.rs`
```rust
use iced_x86::code_asm::*;

fn main() {

    let mut a = CodeAssembler::new(64).unwrap();
    a.sub(10, rax);
}
```

this gives me:
```rust
    Checking yeet v0.1.0 (/home/tornax/yeet)
error[E0277]: the trait bound `iced_x86::code_asm::CodeAssembler: CodeAsmSub<{integer}, _>` is not satisfied
     --> src/main.rs:6:7
      |
6     |     a.sub(10, rax);
      |       ^^^ the trait `CodeAsmSub<{integer}, _>` is not implemented for `iced_x86::code_asm::CodeAssembler`
      |
      = help: the following implementations were found:
                <iced_x86::code_asm::CodeAssembler as CodeAsmSub<AsmMemoryOperand, AsmRegister16>>
                <iced_x86::code_asm::CodeAssembler as CodeAsmSub<AsmMemoryOperand, AsmRegister32>>
                <iced_x86::code_asm::CodeAssembler as CodeAsmSub<AsmMemoryOperand, AsmRegister64>>
                <iced_x86::code_asm::CodeAssembler as CodeAsmSub<AsmMemoryOperand, AsmRegister8>>
              and 17 others
note: required by a bound in `code_asm::fn_asm_pub::<impl iced_x86::code_asm::CodeAssembler>::sub`
     --> /home/tornax/.cargo/registry/src/github.com-1ecc6299db9ec823/iced-x86-1.17.0/src/code_asm/fn_asm_pub.rs:20037:9
      |
20037 |         Self: CodeAsmSub<T, U>,
      |               ^^^^^^^^^^^^^^^^ required by this bound in `code_asm::fn_asm_pub::<impl iced_x86::code_asm::CodeAssembler>::sub`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `yeet` due to previous error
```

## What have you tried?
I've tried to use `AsmMemoryOperand::from` but I'm a little bit unsure if it does what I expect:
```rust
use iced_x86::code_asm::*;

fn main() {

    let mut a = CodeAssembler::new(64).unwrap();
    a.sub(AsmMemoryOperand::from(10), rax).unwrap();
}
```

**Comment by TornaxO7** [comment]
Ok, one of my friends explained it to me.

## TL;DR
The order of the arguments are wrong.

## Long story
As shown in the docs, the [`sub`](https://docs.rs/iced-x86/1.17.0/iced_x86/code_asm/asm_traits/trait.CodeAsmSub.html#tymethod.sub) function has the following signature:
```rust
fn sub(&mut self, op0: T, op1: U) -> Result<(), IcedError>
```
and if you take a look into its [implementations](https://docs.rs/iced-x86/1.17.0/iced_x86/code_asm/asm_traits/trait.CodeAsmSub.html#implementors) you'll see that mostly the register is on the left side.

**Comment by wtfsck** [comment]
That's right, the operands are in the same order as documented by Intel in their documentation, not what the GNU assembler's AT&T syntax uses (wrong order).
